src/libstore/lock.cc

```diff
@@ -39,9 +39,9 @@ struct SimpleUserLock : UserLock
     gid_t gid;
     std::vector<gid_t> supplementaryGIDs;
 
-    uid_t getUID() override { assert(uid); return uid; }
+    uid_t getUID() override { return uid; }
     uid_t getUIDCount() override { return 1; }
-    gid_t getGID() override { assert(gid); return gid; }
```

Comment:
/**
     * Locks do not prevent concurrent access from the root user,
     * so we assert that the root group is not using this. The code using `SimpleUserLock`
     * is required to uphold this invariant.
     */